columnflow/tasks/framework/mixins.py

```diff
@@ -691,10 +691,35 @@ def store_parts(self) -> law.util.InsertableDict:
         # since MLTraining is no CalibratorsMixin, SelectorMixin, ProducerMixin, ConfigTask,
         # all these parts are missing in the `store_parts`
 
-        configs_repr = "__".join(self.configs)
+        configs_repr = "__".join(self.configs[:5])
+
+        if len(self.configs) > 5:
+            configs_repr += f"_{law.util.create_hash(self.configs[5:])}"
+
         parts.insert_after("task_family", "configs", configs_repr)
 
-        # TODO: parts for calib, sel, prod
+        task_array_functions = {
+            "calib": self.calibrators,
+            "sel": tuple((sel,) for sel in self.selectors),
+            "prod": self.producers,
+        }
+
+        for label, fct_names in task_array_functions.items():
+            if all(map(lambda x: x == fct_names[0], fct_names)):
+                # when functions are the same per config, only use them once
+                fct_names = fct_names[0]
+                n_fct_per_config = str(len(fct_names))
+            else:
+                # when functions differ between configs, flatten
+                n_fct_per_config = "".join(str(len(x)) for x in fct_names)
+                fct_names = sum(fct_names[1:], fct_names[0])
```

Comment:
What is the reason for using `sum` in this way here? It seems like it would be adding a tuple to a string, which would raise an exception. Since you only use `fct_names[2:]` below anyway, couldn't this line be dropped completely?

Reply:
Ah, sorry, the entries in `fcn_names` are all tuples, so this is essentially just flattening the list of producers across all configs, right? But then it might be better to use a comprehension for better readability.
```suggestion
                fct_names = tuple(fct_name for fct_names_cfg in fct_names for fct_name in fct_names_cfg)
```